testdata/default/cheats/Toml.t.sol

```diff
@@ -225,48 +299,80 @@ contract ParseTomlTest is DSTest {
         vm._expectCheatcodeRevert("key \".*\" must return exactly one JSON object");
         vm.parseTomlKeys(tomlString, ".*");
     }
+
+    // forge eip712 testdata/default/cheats/Toml.t.sol -R 'cheats=testdata/cheats' -R 'ds-test=testdata/lib/ds-test/src' | grep ^FlatToml
+    string constant schema_FlatToml =
+        "FlatToml(uint256 a,int24[][] arr,string str,bytes b,address addr,bytes32 fixedBytes)";
+
+    // forge eip712 testdata/default/cheats/Toml.t.sol -R 'cheats=testdata/cheats' -R 'ds-test=testdata/lib/ds-test/src' | grep ^NestedToml
+    string constant schema_NestedToml =
+        "NestedToml(FlatToml[] members,AnotherFlatToml inner,string name)AnotherFlatToml(bytes4 fixedBytes)FlatToml(uint256 a,int24[][] arr,string str,bytes b,address addr,bytes32 fixedBytes)";
+
+    function test_parseTomlType() public {
+        string memory readToml = vm.readFile("fixtures/Toml/nested_toml_struct.toml");
+        NestedToml memory data = readToml.deserializeNestedToml();
+        assertEq(data.members.length, 2);
+
+        FlatToml memory expected = FlatToml({
+            a: 200,
+            arr: new int24[][](0),
+            str: "some other string",
+            b: hex"0000000000000000000000000000000000000000",
+            addr: 0x167D91deaEEE3021161502873d3bcc6291081648,
+            fixedBytes: 0xed1c7beb1f00feaaaec5636950d6edb25a8d4fedc8deb2711287b64c4d27719d
+        });
+
+        assertEq(keccak256(abi.encode(data.members[1])), keccak256(abi.encode(expected)));
+        assertEq(bytes32(data.inner.fixedBytes), bytes32(bytes4(0x12345678)));
+
+        FlatToml[] memory members = TomlStructs.deserializeFlatTomlArray(readToml, ".members");
+
+        assertEq(keccak256(abi.encode(members)), keccak256(abi.encode(data.members)));
+    }
 }
 
 contract WriteTomlTest is DSTest {
     Vm constant vm = Vm(HEVM_ADDRESS);
 
-    string json1;
-    string json2;
+    string toml1;
+    string toml2;
 
     function setUp() public {
-        json1 = "example";
-        json2 = "example2";
+        toml1 = "example";
+        toml2 = "example2";
     }
 
-    struct simpleJson {
+    struct simpleToml {
         uint256 a;
         string b;
     }
 
-    struct notSimpleJson {
+    struct notSimpleToml {
         uint256 a;
         string b;
-        simpleJson c;
+        simpleToml c;
     }
 
     function test_serializeNotSimpleToml() public {
-        string memory json3 = "json3";
+        string memory toml3 = "toml3";
         string memory path = "fixtures/Toml/write_complex_test.toml";
-        vm.serializeUint(json3, "a", uint256(123));
-        string memory semiFinal = vm.serializeString(json3, "b", "test");
-        string memory finalJson = vm.serializeString(json3, "c", semiFinal);
-        console.log(finalJson);
-        vm.writeToml(finalJson, path);
```

Comment:
I originally used the `json` naming here to indicate that during operations users are operating on JSON data, it only becomes TOML once written

I would prefer this to be reverted but open to feedback

Reply:
That makes sense, I'll revert the changes

Reply:
I've kept the `struct` renaming because I think it makes more sense, but I'm open to change :)